scripts/recovery/protect-and-retire-noncanonical-worktrees.sh

```
@@ -0,0 +1,392 @@
+  branch="$(git -C "$repo" branch --show-current)"
+  if [[ -z "$branch" ]] || integration_branch "$branch"; then
+    branch="$(make_recovery_branch "$repo" "$label")"
+  fi
+  printf '%s\n' "$branch"
+}
+
+remote_has_head() {
+  local repo="$1"
+  local head
+  head="$(git -C "$repo" rev-parse HEAD)"
+  git -C "$repo" ls-remote --heads origin | grep -q "^$head[[:space:]]"
```

Shellcheck (0.11.0)

[error] Use braces when expanding arrays, e.g. ${array[idx]} (or ${var}[.. to quiet).

(SC1087)